scripts/configurator/parser/cmake_var.py

```diff
@@ -1,7 +1,104 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"([A-Z_]+)")
+
+   def __init__(self, comment: str) -> None:
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw_to_parse: Mapping[str, function] = {
+         "REALTYPE" : self.parse_realtype,
+         "DEPENDS_ON" : self.parse_depends_on,
+         "GROUP" : self.parse_group
+      }
+
+      self.kw_to_setter = {
+         "REALTYPE": self.get_setter_realtype,
+         "DEPENDS_ON": self.get_setter_depends_on,
+         "GROUP": self.get_setter_group,
+      }
+      self.kw_to_var = {
+         "REALTYPE": self.realtype,
+         "DEPENDS_ON": self.depends_on,
+         "GROUP": self.group,
+      }
+      for i, line in enumerate(comment.splitlines()):
+         match = self.keyword_expr.match(line)
+         if match:
+            key = match.groups()[0]
+            if i == 0:
+               if key in self.kw_to_parse:
+                  msg = """WARNING: First line uses a reserved keyword: {}.
+                  This will be ignored."""
+                  stderr.write(msg.format(key))
+               continue
+            elif key not in self.kw_to_parse:
+               msg = "{} is not a valid metadata keyword."
+               raise ValueError(msg.format(key))
+            elif self.kw_to_var[key] is not None:
+               msg = "Key {} has already a value associated to itself."
+               raise ValueError(msg.format(key))
+            parse_function = self.kw_to_parse[key]
+            not_matched = line[len(key):].lstrip()
+            get_setter= self.kw_to_setter[key]
```

Comment:
If you put the closures in the dict, as I suggested above, here you'll just do:

```
setter = self.kw_to_setter[key]
```

Reply:
perfect. yes, that's defintely better